.github/workflows/unit-tests.yaml

```diff
@@ -3,14 +3,16 @@ name: Unit tests
 on:
   push:
     branches: [ "main" ]
-  pull_request:
+  pull_request_target:
     branches: [ "main" ]
 
 jobs:
   build:
     runs-on: ubuntu-latest
     steps:
     - uses: actions/checkout@v4
+      with:
+        ref: ${{ github.event.pull_request.head.sha }}
```

Comment:
We can drop this too if switching back to `pull_request` event (and flip the `if` below back`).